routes/api/contacts.js

```diff
@@ -1,25 +1,100 @@
-const express = require('express')
+const express = require("express");
+const Joi = require("joi");
+const {
+  listContacts,
+  getContactById,
+  removeContact,
+  addContact,
+  updateContact,
+} = require("../../models/contacts");
 
-const router = express.Router()
+const router = express.Router();
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const schema = Joi.object({
+  name: Joi.string().required().messages({
+    "any.required": "Missing required name field",
+  }),
+  email: Joi.string().email().required().messages({
+    "any.required": "Missing required email field",
+    "string.email": "Email must be a valid email address",
+  }),
+  phone: Joi.string().required().messages({
+    "any.required": "Missing required phone field",
+  }),
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  try {
+    const contacts = await listContacts();
+    res.json(contacts);
+  } catch (error) {
+    next(error);
+  }
+});
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+  try {
+    const { contactId } = req.params;
+    const contact = await getContactById(contactId);
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+    if (contact) {
+      res.json(contact);
+    } else {
+      res.status(404).json({ message: "Not Found" });
+    }
+  } catch (error) {
+    next(error);
+  }
+});
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  try {
+    const { error } = schema.validate(req.body);
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
+    } else {
+      const newContact = await addContact(req.body);
+      res.status(201).json(newContact);
+    }
+  } catch (error) {
+    next(error);
+  }
+});
 
-module.exports = router
+router.delete("/:contactId", async (req, res, next) => {
+  try {
+    const { contactId } = req.params;
+    const removedContact = await removeContact(contactId);
+
+    if (removedContact) {
+      res.status(200).json({ message: "Contact Deleted" });
+    } else {
+      res.status(404).json({ message: "Not Found" });
+    }
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.put("/:contactId", async (req, res, next) => {
+  try {
+    const { contactId } = req.params;
+    const { error } = schema.validate(req.body);
+
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
+    }
+
+    const updatedContact = await updateContact(contactId, req.body);
+
+    if (updateContact) {
+      res.status(200).json(updatedContact);
+    } else {
+      res.status(404).json({ message: "Not Found" });
+    }
```

Comment:
Poczytaj trochę o sposobie pisania `early return statement` pozwala ona np na to aby było mniej if else. 
Np:
```
if (updateContact) return res.status(200).json(updateContact)

res.status(404).jsn({message: "Not Found"})
```

Reply:
W celu zobaczenia innych sposobów pisania ;)